src/app/auth/auth.component.ts

```diff
@@ -105,12 +104,7 @@ export class AuthComponent implements OnInit, OnDestroy {
 
     const { oauthCode, state } = event.data;
 
-    if (!oauthCode) {
-        return;
-    }
-
-    if (!this.authService.isReturnedStateSame(state)) {
```

Comment:
We cannot remove the `this.authService.isReturnedStateSame(state)` call.
This is needed for security purposes, and some changes are needed here.

Within `AuthService.generateStateString`, we need to store the generated `state` in **session storage** (not local storage). Within the `onMessage` method, we need to keep the `if (!this.authService.isReturnedStateSame(state))` check. 

Within `AuthService.isReturnedStateSame`, we will need to recover the original `state` string from session storage, before validating it against the state obtained from the redirect.

The `state` string is a random string that is meant to protect CATcher from [CSRF attacks](https://en.wikipedia.org/wiki/Cross-site_request_forgery). Take a look at [this site](https://auth0.com/docs/protocols/state-parameters) to read about the purpose of `state`.

Reply:
Thanks for the information and suggestion!
Will make the necessary changes